### PR TITLE
Basic Lobstrosity

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

I'm slowly chipping away at mining mobs. These ones also got some new sprites because the old ones were a bit weird except when facing South.
image
Arctic Lobstrosities are now hairy to give them a little more visual distinction from Lavaland ones.

In terms of behaviour, they're now a little faster and can charge you from further away.
They will _only_ attack players who are incapacitated in some way (primarily from being hit by their charge, but could be from a Goliath or something too) and will otherwise keep their distance until they can charge again. They move slower for a short duration after charging though, so you have time to slap them a bit.

If a Lobstrosity downs you then it will try to snip off one of your arms, then retreat in order to eat it.
Obviously nobody likes losing an arm, but this does give you an opportunity to get away while it is distracted? Funnily enough the way our health system works means that sometimes losing that arm actually takes you out of soft crit so you can stumble back to the station for a replacement (or try to wrestle yours back?)

All of these things are achievable also by a player if you make one sapient, they will pull arms off mobs they attack which are in crit and can eat arms if they see them lying around if they want.
I added an element to let you dismember people with your bare hands, maybe someone evil can use it to add a beheading attack some day.

Here's a video of their new behaviours:
https://www.youtube.com/watch?v=9eKxsH7hD7Q

## Why It's Good For The Game

Gives mobs more character.
Reduces our list of frozen simple mobs.
Replaces some ugly side sprites.
Medbay enrichment?

## Changelog

:cl:
refactor: Lobstrosities are now basic mobs and have different AI behaviour. Please report anything which seems like it shouldn't be happening.
add: Lobstrosities will now only opportunistically attack things they have knocked over with their charge, and are otherwise timid.
add: Lobstrosities are hungry for fingers and will steal one of your arms if they defeat you in combat, although this gives you time to crawl away.
sprite: New sprites for Lobstrosities.
/:cl:

### CODE REVIEW

**Comment by GuillaumePrata** [comment]
> Obviously nobody likes losing an arm, but this does give you an opportunity to get away while it is distracted? Funnily enough the way our health system works means that sometimes losing that arm actually takes you out of soft crit

Not only is this behavior funny, but it also "removes" the worst part of our mining desing.
Dying as miner is overly punishing (Ignoring the round removals) with how long it takes for you to be found and revived. Now if you are lucky it is replaced with a fair punishment, losing a limb, but you can just walk back to the station yourself and get it fixed while cutting on the annoying "wait and hope someone comes help me" period of failures as a miner.

This honestly should be a design goal moving forward whenever it makes sense.

**Comment by carlarctg** [comment]
> > Obviously nobody likes losing an arm, but this does give you an opportunity to get away while it is distracted? Funnily enough the way our health system works means that sometimes losing that arm actually takes you out of soft crit
> 
> Not only is this behavior funny, but it also "removes" the worst part of our mining desing. Dying as miner is overly punishing (Ignoring the round removals) with how long it takes for you to be found and revived. Now if you are lucky it is replaced with a fair punishment, losing a limb, but you can just walk back to the station yourself and get it fixed while cutting on the annoying "wait and hope someone comes help me" period of failures as a miner.
> 
> This honestly should be a design goal moving forward whenever it makes sense.

Agreed, it makes me wonder of similar replacements for legions, watchers, goliaths.

**Comment by atlasle** [comment]
> Agreed, it makes me wonder of similar replacements for legions, watchers, goliaths.
Goliaths need like lairs that they pull you to, that would also home some loot (maybe like goliath eggs or ambergris) that would reward other miners for saving you (also we really need to be able to attach a health sensor to a GPS to activate a distress beacon that would make the signal blink red or something)

**Comment by SimplySmiley123** [comment]
> > Agreed, it makes me wonder of similar replacements for legions, watchers, goliaths.
> 
> Goliaths need like lairs that they pull you to, that would also home some loot (maybe like goliath eggs or ambergris) that would reward other miners for saving you (also we really need to be able to attach a health sensor to a GPS to activate a distress beacon that would make the signal blink red or something)

This would honestly be an amazing thing to have.
Especially when it comes to miners who forget to name their GPS something to let the other miners or the singular paramedic that gives a damn which GPS signal to look for.

**Comment by Jacquerel** [comment]
I honestly don't have clear ideas for equivalents for any other mob but I will try to keep it in mind, it's an interesting concept.

File: code/datums/ai/basic_mobs/basic_ai_behaviors/run_away_from_target.dm [hunk 1]
```diff
@@ -16,6 +18,8 @@
 
 /datum/ai_behavior/run_away_from_target/perform(seconds_per_tick, datum/ai_controller/controller, target_key, hiding_location_key)
 	. = ..()
+	if (!controller.blackboard[BB_BASIC_MOB_FLEEING])
```
Comment (san7890) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/run_away_from_target.dm:21]: i forgot, is fleeing a boolean or does it resolve to the mob it's fleeing from? use `isnull()` if it's the latter but i could just be smallbrained rn

File: code/datums/mapgen/Cavegens/IcemoonCaves.dm [hunk 2]
```diff
@@ -6,7 +6,7 @@
 	weighted_mob_spawn_list = list(/mob/living/simple_animal/hostile/asteroid/wolf = 50, /obj/structure/spawner/ice_moon = 3, \
 						  /mob/living/simple_animal/hostile/asteroid/polarbear = 30, /obj/structure/spawner/ice_moon/polarbear = 3, \
 						  /mob/living/simple_animal/hostile/asteroid/hivelord/legion/snow = 50, /mob/living/simple_animal/hostile/asteroid/goldgrub = 10, \
-						  /mob/living/simple_animal/hostile/asteroid/lobstrosity = 15)
+						  /mob/living/basic/mining/lobstrosity = 15)
```
Comment (san7890) [inline code/datums/mapgen/Cavegens/IcemoonCaves.dm:9]: A) alphabetize
B) this list sucks can you properly multiline it

File: code/modules/fishing/sources/source_types.dm [hunk 3]
```diff
@@ -71,7 +71,7 @@
 	fish_table = list(
 		FISHING_DUD = 5,
 		/obj/item/fish/chasm_crab/ice = 15,
-		/mob/living/simple_animal/hostile/asteroid/lobstrosity = 1,
+		/mob/living/basic/mining/lobstrosity = 1,
```
Comment (san7890) [inline code/modules/fishing/sources/source_types.dm:74]: alphabetize

File: code/modules/fishing/fish/chasm_detritus.dm [hunk 4]
```diff
@@ -22,16 +22,16 @@
...
 		BODIES_ONLY = list(
 			/obj/effect/mob_spawn/corpse/human/skeleton = 3,
-			/mob/living/simple_animal/hostile/asteroid/lobstrosity/lava = 1,
+			/mob/living/basic/mining/lobstrosity/lava = 1,
 		),
 		NO_CORPSES = list(
 			/obj/item/stack/ore/slag = 10,
 			/obj/item/stack/sheet/bone = 14,
-			/mob/living/simple_animal/hostile/asteroid/lobstrosity/lava = 1,
+			/mob/living/basic/mining/lobstrosity/lava = 1,
```
Comment (san7890) [inline code/modules/fishing/fish/chasm_detritus.dm:34]: alphabetize all three of these lists

File: code/modules/mob/living/basic/lavaland/goliath/goliath_trophy.dm [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/// Mining crusher trophy from a goliath. Increases damage as your health decreases.
+/obj/item/crusher_trophy/goliath_tentacle
+	name = "goliath tentacle"
+	desc = "A sliced-off goliath tentacle. Suitable as a trophy for a kinetic crusher."
+	icon_state = "goliath_tentacle"
+	denied_type = /obj/item/crusher_trophy/goliath_tentacle
+	bonus_value = 2
+	var/missing_health_ratio = 0.1
+	var/missing_health_desc = 10
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/goliath/goliath_trophy.dm:9]: would you mind dmdoccing while we're here

File: code/modules/mob/living/basic/lavaland/lobstrosity/lobstrosity.dm [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+/// Lavaland lobster variant, it basically just looks different
+/mob/living/basic/mining/lobstrosity/lava
+	name = "chasm lobstrosity"
+	desc = "Twitching crustaceans boiled red by the sulfurous fumes of the chasms in which they lurk. They have a particular taste for fingers."
+	icon_state = "lobstrosity"
+	icon_living = "lobstrosity"
+	icon_dead = "lobstrosity_dead"
+
+/// Charge a long way, knock down for longer, and perform an instant melee attack
+/datum/action/cooldown/mob_cooldown/charge/basic_charge/lobster
```
Comment (san7890) [inline code/modules/mob/living/basic/lavaland/lobstrosity/lobstrosity.dm:67]: lober

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/basic/lavaland/lobstrosity/lobstrosity.dm (hunk 6 above)
Comment (san7890) [reply code/modules/mob/living/basic/lavaland/lobstrosity/lobstrosity.dm:67]: no change

File: code/datums/ai/basic_mobs/basic_ai_behaviors/run_away_from_target.dm (hunk 1 above)
Comment (Jacquerel) [reply code/datums/ai/basic_mobs/basic_ai_behaviors/run_away_from_target.dm:21]: It's just a boolean, don't worry I was thinking of you.

**Comment by Jacquerel** [comment]
Some of those lists I ordered by probability _and_ alphabetically instead based on my whim, vibes-based sorting.
I looked at the ai defines file, couldn't decide how to split it, and decided not to do it.

**Comment by LSommers** [comment]
> Goliaths need like lairs that they pull you to, that would also home some loot (maybe like goliath eggs or ambergris) that would reward other miners for saving you (also we really need to be able to attach a health sensor to a GPS to activate a distress beacon that would make the signal blink red or something)

goliath egg that you can hatch into a useless but very cute baby goliath pet
(several assistants rush to lavaland and immediately die trying to get the new cosmetic item)

**Comment by atlasle** [comment]
> goliath egg that you can hatch into a useless but very cute baby goliath pet
> 
> (several assistants rush to lavaland and immediately die trying to get the new cosmetic item)

Our brains are on the same wavelength

**Pull request merged**